purchases/src/main/kotlin/com/revenuecat/purchases/HTTPClient.kt

```diff
@@ -117,6 +105,23 @@ internal class HTTPClient(
         return result
     }
 
+    private fun Map<String, Any?>.convert(): JSONObject {
+        val mapWithoutInnerMaps = mapValues { (_, value) ->
+            value.tryCast<Map<String, Any?>>(ifSuccess = { convert() })
+        }
```

Comment:
❤️ kotlin functional programming. 
Maybe we could leave a comment here explaining _why_ this is needed, for future reference? So that a future dev doesn't come here, see this and say "wait, this isn't needed in modern android" and just kills the line

Reply:
lol, yeah good point, it took me a while to get this to work actually

Reply:
Added a comment here